Set list of uplink-purchasable items for Gang Mode

Ikarrus opened the pull request:
This excludes certain uplink items that can be purchased during gang mode. Most notably emags (OP with borgs), space suits and chameleon projector (No hiding), the singularity beacon and certain spy-like weapons (Not appropriate).
But this also allows gang bosses to purchase items that were formerly only available to nuke ops, such as their extensive arsenal of weapons.

Ikarrus committed:
Custom list of uplink-purchasable items for Gang

This excludes certain uplink items that can be purchased during gang mode. Most notably emags (OP with borgs), space suits (No hiding), chameleon projector (No hiding), the singularity beacon and certain spy-like weapons (Not appropriate).
But this also allows gang bosses to purchase items that were formerly only available to nuke ops, such as their weapons.

Miauw62 commented on code/datums/uplink_item.dm:
```diff
@@ -684,7 +689,7 @@ var/list/uplink_items = list()
 	cost = 20
 	item = /obj/item/weapon/storage/box/syndicate
-	excludefrom = list(/datum/game_mode/nuclear)
+	excludefrom = list(/datum/game_mode/nuclear,/datum/game_mode/gang)
```
of note is that because of the way uplinks work, the surplus crate actually only dispenses items from the current mode.

Ikarrus commented:
@Miauw62 neat, I didn't know that. I'll allow it then.

Ikarrus committed:
Unexcluded surplus crates

KorPhaeron commented:
Would a new nerfed emag that doesn't work on borgs be possible to add for them?

Just so they have a tool for easy break ins

Ikarrus commented:
Doorjack maybe. When I have time.

SconesC commented:
C4 is allll the break-in you need

Have you made sure that the weapons will spawn with the correct pin config for usage by the gang?

Ikarrus commented:
What the fuck is a pin oh god new stuff

KorPhaeron commented:
C4 is slow and loud.

Also limited use and good luck getting into areas with multiple layers of doors with it

SconesC commented:
@Ikarrus Make the guns spawn with an electronic firing pin instead of syndicate - That, or flag the gang members as nuke ops for the weapons whitelist

Ikarrus committed:
More adjustments
- added unrestricted-use guns
- excluded fake toy guns
- Unexcluded throwing stars

Miauw62 commented:
fairly sure that the nuke op pin check checks for an implant in the ops.

Ikarrus commented:
I added unrestricted subtypes of the weapons and have it spawn those.

Miauw62 commented on code/modules/projectiles/guns/projectile/automatic.dm:
```diff
@@ -88,6 +88,9 @@
 	pin = /obj/item/device/firing_pin/implant/pindicate
 
+/obj/item/weapon/gun/projectile/automatic/c20r/unrestricted
+	pin = /obj/item/device/firing_pin
```
fairly sure this should be new /obj/item/device/firing_pin(src)

RemieRichards replied:
No @Miauw62 this is a definition.
You cant have objects in a definition.

Also 2 lines above ```pin = /obj/item/device/firing_pin/implant/pindicate```

Ignore Miauw, he's a silly.

Miauw62 replied:
oh w/e

you can totally have objects in a definition btw.
it's done in tons of places.

RemieRichards replied:
My mistake, it is done in places now that I mention it, but not here.
The point I was making erroneously was just constant expressions.

In any case, what Ikky has done here is fine.

Miauw62 commented on code/modules/projectiles/guns/projectile/automatic.dm:
```diff
@@ -175,6 +178,9 @@
 	pin = /obj/item/device/firing_pin/implant/pindicate
 
+/obj/item/weapon/gun/projectile/automatic/m90/unrestricted
+	pin = /obj/item/device/firing_pin
```
see above

RemieRichards replied:
Except don't, because he's a silly.
